rasa/shared/core/domain.py

```diff
@@ -698,24 +698,24 @@ def num_states(self):
         return len(self.input_states)
 
     @rasa.shared.utils.common.lazy_property
-    def retrieval_intent_templates(self) -> Dict[Text, List[Dict[Text, Any]]]:
-        """Return only the templates which are defined for retrieval intents"""
+    def retrieval_intent_responses(self) -> Dict[Text, List[Dict[Text, Any]]]:
```

Comment:
we can't just rename this method and the following. We need to deprecate the old versions.

Reply:
Hm I think for the deprecation discussion I need to understand the relationship with rasa.nlu, rasa.core and the API. Should we hop on a quick call?

Reply:
Fixed